standalone/src/main/scala/io/delta/standalone/internal/data/CloseableParquetDataIterator.scala

```
@@ -116,12 +126,53 @@ private[internal] case class CloseableParquetDataIterator(
   }
 
   /**
-   * Requires that `dataFilePathsIter.hasNext` is true.
+   * Requires that `dataFilePathsAndPartitionsIter.hasNext` is true.
    *
-   * @return the iterable for the next data file in `dataFilePathsIter`, not null
+   * @return the iterable for the next data file in `dataFilePathsAndPartitionsIter`, not null
    */
   private def readNextFile: ParquetIterable[RowParquetRecord] = {
+    val (nextDataFilePath, nextPartitionVals) = dataFilePathsAndPartitionsIter.next()
+
+    partitionValues = Map()
+
+    if (null != nextPartitionVals) {
+      nextPartitionVals.foreach { case (fieldName, value) =>
+        if (value == null || "null" == value) {
+          partitionValues += (fieldName -> null)
+        } else {
+          val schemaField = schema.get(fieldName)
+          if (schemaField != null) {
+            val decodedFieldValue = decodePartition(schemaField.getDataType, value)
+            partitionValues += (fieldName -> decodedFieldValue)
+          }
```

it's better to throw an exception for the invalid state (`schemaField==null`).